### ISSUE TITLE
Site Does not Load on Latest Electron

### ISSUE THREAD

**Issue by OrangeDrangon** [OrangeDrangon/android-messages-desktop#426]
When opening the application with an updated electron (26). It hangs on a white screen. With these errors in console.

![image](https://github.com/OrangeDrangon/android-messages-desktop/assets/25403517/101aaccd-2f79-4630-904b-1324ac716914)

**Comment by OrangeDrangon** [comment]
The most modern version it does load in is 21.

**Comment by george-dy** [comment]
I've never built anything with Electron, but it looks like it might have something to do with CSP? I just wanted to add some links I found while searching, though I don't know if any of them are actually useful.

https://csplite.com/csp219/
https://stackoverflow.com/a/74201095
https://stackoverflow.com/questions/62081028/this-document-requires-trustedscripturl-assignment

**Comment by LanikSJ** [comment]
I've gotten the latest Electron to play nice using the TypeScript files from this commit: https://github.com/ElementalWarrior/android-messages-desktop/commit/e7f2e0215318a5dc4e79a881330a1886cf06eac5#diff-11db236acd6b02b1229e7d325e9bf8edb521032e6388f68b6a884ec24bbe291f

You can test my fork if you like.  I'm using the app right now and it works great, builds etc.

**Comment by OrangeDrangon** [comment]
fixed in #472

**Comment by ElementalWarrior** [comment]
So I was actually curious about why upgrading didn't work before. And you were right, the CSP stuff was blocking it. I played around.

The below would actually fix it without turning on context isolation. The problem lies in the google web requests setting `require-trusted-types-for ...` in the CSP header. But no matching trusted types policy names are defined. So removing them before the browser processes them actually resolves the issue.

I don't think the PR I made and was merged was bad, in fact its probably safer. Albeit it may be more painful development wise if for whatever reason other node API's need to be used in the future.

```


    // strip out require-trusted-types-for. In electron@22.3.6 they started enforcing it.
    mainWindow.webContents.session.webRequest.onHeadersReceived((details, callback) => {
      let csp = []
      for (let ele of (details.responseHeaders??{})["content-security-policy"] ?? []) {
        csp.push(ele.replace(/require-trusted-types-for[^;]+;?/, ""))
      }
      
      callback({
        responseHeaders: {
          ...details.responseHeaders,
          'content-security-policy': csp
        }
      })
    })
```
